tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/impl/WeightsBroadcastOps.java

```diff
@@ -0,0 +1,186 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.metrics.impl;
+
+import org.tensorflow.Operand;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.op.Op;
+import org.tensorflow.op.Ops;
+import org.tensorflow.types.TBool;
+import org.tensorflow.types.TInt32;
+import org.tensorflow.types.family.TNumber;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.tensorflow.framework.utils.CastHelper.cast;
+
+public class WeightsBroadcastOps {
+
+  private static final String ASSERT_BROADCASTABLE_ERROR_PREFIX =
+      "weights can not be broadcast to values.";
+
+  /**
+   * Asserts that `weights` can be broadcast to `values`
+   *
+   * @param tf the TensorFlow Ops
+   * @param weights `Tensor` of weights.
+   * @param values `Tensor` of values to which weights are applied.
+   * @return `Operation` raising `InvalidArgumentError` if `weights` has incorrect shape. `no_op` if
+   *     static checks determine `weights` has correct shape.
+   * @param <T> the type of weights and values
+   * @throws IllegalArgumentException If static checks determine `weights` has incorrect shape.
+   */
+  @SuppressWarnings("unchecked")
```

Comment:
We can eliminate the need for `@SuppressWarnings("unchecked")` by using a different idiom for constructing empty lists:
~~~Java
            .withControlDependencies(Collections.emptyList())
~~~

Reply:
OK, changed` Collections.emptyList()` and removed  `@SuppressWarnings("unchecked")`